### PR TITLE
Generate doxygen

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
N/A

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Doc

**What is the current behavior?** *(You can also link to an open issue here)*
No doxygen is generated with the project

**What is the new behavior (if this is a feature change)?**
Add a `doxygen` target to generate doxygen documentation for the public API. The documentation is installed in the `share/doc` folder.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,7 +44,7 @@ enable_testing()
 # https://github.com/boostorg/test/commit/d63379a9b3ad33d263d55c119681047c1d28547c
 if ((DEFINED Boost_VERSION_MACRO AND ${Boost_VERSION_MACRO} GREATER 106000) OR (${Boost_VERSION} GREATER 106000))
     set(BOOST_ARGS_SEPARATOR "--")
-endif()
+endif ()
```
Comment (gautierbureau) [inline CMakeLists.txt]: Why ?

File: README.md [hunk 2]
```diff
@@ -50,3 +50,28 @@ $> cmake .. -DCMAKE_INSTALL_PREFIX=<PREFIX> -DCMAKE_BUILD_TYPE=<BUILD_TYPE>
 $> make -j
 $> make install
 ```
+
+### Checkstyle
+This project uses [clang-tidy](https://clang.llvm.org/extra/clang-tidy/) to verify the code style. This tool is provided with
+the clang extra tools. To enable the code style checking, add `-DCMAKE_CXX_CLANG_TIDY=clang-tidy` to the cmake command.
```
Comment (gautierbureau) [inline README.md]: why newline ?

File: CMakeLists.txt (hunk 1 above)
Comment (mathbagu) [reply CMakeLists.txt]: To follow the coding style. Do you prefer I revert this modification and create a pull request for this specific change?

File: README.md (hunk 2 above)
Comment (mathbagu) [reply README.md]: To make the file more readable. What do you suggest: to keep everything on a single line (246 characters)? I think it's really hard to maintain, and my IDE has a feature to automatically wrap long line (120 characters)

File: CMakeLists.txt (hunk 1 above)
Comment (gautierbureau) [reply CMakeLists.txt]: ok

File: README.md (hunk 2 above)
Comment (gautierbureau) [reply README.md]: ok

**Pull request merged**